bin/repair-fixable-orphans.py

```diff
@@ -0,0 +1,165 @@
+    "--env",
+    default="uat",
+    required=False,
+    help="Environment to run against, eg 'production, 'uat', etc (default: '%(default)s')",
+)
+parser.add_argument(
+    "--pgcli",
+    default="/usr/bin/psql",
+    required=False,
+    help=f"Postgres client path (default: %(default)s)",
+)
+args = parser.parse_args()
+
+# Keep synced with other scripts
+GLOSS_ID_COLUMN = "Gloss ID"
+GLOSS_COLUMN = "Gloss"
+GLOSS_PUBLIC_COLUMN = "Gloss public"
+GLOSS_VIDEO_COLUMN = "Suggested Video key"
+GLOBAL_COLUMN_HEADINGS = [
+    GLOSS_ID_COLUMN,
+    GLOSS_COLUMN,
+    GLOSS_PUBLIC_COLUMN,
+    GLOSS_VIDEO_COLUMN,
+]
+
+# Other globals
+CSV_DELIMITER = ","
+FAKEKEY_PREFIX = "this_is_not_a_key_"
+DATABASE_URL = os.getenv("DATABASE_URL", "")
+PGCLI = args.pgcli
+AWS_S3_BUCKET = f"nzsl-signbank-media-{args.env}"
+
+
+def pg_cli(args_list):
+    try:
+        return subprocess.run(
+            [PGCLI, "-c"] + args_list + [f"{DATABASE_URL}"],
+            env=os.environ,
+            capture_output=True,
+            check=True,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        print(f"Error: subprocess.run returned code {e.returncode}", file=sys.stderr)
+        print(e.cmd, file=sys.stderr)
+        print(e.stdout, file=sys.stderr)
+        print(e.stderr, file=sys.stderr)
+        exit()
+
+
+# Returns a list of dictionaries, one for each CSV row
+def read_csv(csv_filename):
+    if csv_filename == "-":
+        f = sys.stdin.read().splitlines()
+    else:
+        f = open(csv_filename, "r")
+    return csv.DictReader(f)
+
+
+def process_csv():
+    main_video_type = FieldChoice.objects.filter(
+        field="video_type", english_name="main"
+    ).first()
+
+    csv_rows = read_csv(args.csv_filename)
+    for csv_row in csv_rows:
+        gloss_id = csv_row[GLOSS_ID_COLUMN]
+        gloss_idgloss = csv_row[GLOSS_COLUMN]
+        video_key = csv_row[GLOSS_VIDEO_COLUMN]
+        print(CSV_DELIMITER.join([gloss_id, gloss_idgloss, video_key]))
+        gloss_id = int(gloss_id)
+
+        try:
+            gloss = Gloss.objects.get(id=gloss_id)
+        except ObjectDoesNotExist as e:
+            print(e)
+            continue
+
+        try:
+            GlossVideo.objects.get(videofile=video_key)
+            print(f"Ignoring: GlossVideo already exists: {video_key}")
+            continue
+        except ObjectDoesNotExist:
+            pass
+
+        gloss_video = GlossVideo(
+            gloss=gloss,
+            dataset=gloss.dataset,
+            videofile=video_key,
+            title=video_key,
+            version=0,
+            is_public=False,
+            video_type=main_video_type,
+        )
+        print(gloss)
+        print(gloss_video)
+        # At this point we complete the repair
+        # We use bulk_create() because we cannot allow save() to run
+        if len(GlossVideo.objects.bulk_create([gloss_video])) < 1:
+            print(f"Error: could not create {gloss_video}")
```

Comment:
Suggestion (non-blocking): Consider a `dry-run` cli flag that defaults to `True`

The purpose of the flag to allow you to run the script end to end to _see_ what it will do without actually making the DB changes. Defaulting it `True` means you have to make an explicit decision to run it for real - it is harder to accidentally make changes to the DB.

Reply:
Good idea. I'll rename and reverse the sense of the flag.